### PR TITLE
save snapshots after we moved the resources

### PR DESCRIPTION
**Pull request by tgummerer**
Once we moved the resources, we also want to save the snapshot.  Do so here.

We try to save the destination snapshot first, so resources don't get orphaned, as it's probably slightly easier for users to delete resources form the source snapshot than it is for them to import them again in the destination. Ideally this would be an atomic operation, but I don't think we have the means to do that right now.

### CODE REVIEW

File: pkg/cmd/pulumi/state_move.go [hunk 1]
```diff
@@ -146,6 +146,21 @@ func (cmd *stateMoveCmd) Run(
...
+		return err
+	}
+	if rootStack == nil {
+		projectName, ok := dest.Ref().Project()
+		if !ok {
+			return errors.New("failed to get project name of source stack")
+		}
+		rootStack = stack.CreateRootStackResource(
+			dest.Ref().Name().Q(), tokens.PackageName(projectName))
+		destSnapshot.Resources = append(destSnapshot.Resources, rootStack)
```
Comment (Frassle) [inline pkg/cmd/pulumi/state_move.go:161]: Shouldn't this always be prepended? Root stack should be the first resource.

File: pkg/cmd/pulumi/state_move.go [hunk 2]
```diff
@@ -185,12 +191,32 @@ func (cmd *stateMoveCmd) Run(
...
+		return fmt.Errorf(`failed to verify integrity of source snapshot: %w
+
+This is a bug! We would appreciate a report: https://github.com/pulumi/pulumi/issues/`, err)
+	}
+
+	// We're saving the destination snapshot first, so that if saving a snapshot fails
+	// the resources will always still be tracked.  If the source snapshot fails the user
+	// will have to manually remove the resources from the source stack.
+	//
+	// TODO: give a better error message and instructions for how the user can fix this
```
Comment (Frassle) [inline pkg/cmd/pulumi/state_move.go:210]: Can we TODO this in this PR?

Comment (tgummerer) [reply pkg/cmd/pulumi/state_move.go]: Done :heavy_check_mark:

File: pkg/cmd/pulumi/state_move.go (hunk 1 above)
Comment (tgummerer) [reply pkg/cmd/pulumi/state_move.go]: Yeah, I think prepending makes sense.  Done :heavy_check_mark:

**Review by Frassle — APPROVED** [review]

**Pull request merged**